File: 58-length-of-last-word/58-length-of-last-word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        n = len(s)
        j = n - 1
        while (j >= 0) and (s[j] == " "):
            j -= 1
        i = j
        while (i >= 0) and (s[i] != " "):
            i -= 1
        if i < 0:
            return j + 1
        return j - i

File: 58-length-of-last-word/58-length-of-last-word.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        int n = s.length();
        int j = n - 1;
        while ((j >= 0) and (s[j] == ' ')) {
            j--;
        }
        int i = j;
        while ((i >= 0) and (s[i] != ' ')) {
            i--;
        }
        if (i < 0) {
            return j + 1;
        }
        return j - i;
    }
};